src/main/org/firebirdsql/gds/ng/wire/SrpClient.java

```diff
@@ -0,0 +1,222 @@
+    private static final BigInteger N = new BigInteger("E67D2E994B2F900C3F41F08F5BB2627ED0D49EE1FE767A52EFCD565CD6E768812C3E1E9CE8F0A8BEA6CB13CD29DDEBF7A96D4A93B55D488DF099A15C89DCB0640738EB2CBDD9A8F7BAB561AB1B0DC1C6CDABF303264A08D1BCA932D1F1EE428B619D970F342ABA9A65793B8B2F041AE5364350C16F735F56ECBCA87BD57B29E7", 16);
+    private static final BigInteger g = new BigInteger("2");
+
+    private static final BigInteger k = new BigInteger("1277432915985975349439481660349303019122249719989");
+
+    private BigInteger publicKey;   /* A */
+    private BigInteger privateKey;  /* a */
+    private byte[] sessionKey;      /* K */
+
+
+    private static class KeyPair {
+        private BigInteger pub, secret;
+        KeyPair(BigInteger pub, BigInteger secret) {
+            this.pub = pub;
+            this.secret = secret;
+        }
+        private BigInteger getPublicKey() {
+            return pub;
+        }
+        private BigInteger getPrivateKey() {
+            return secret;
+        }
+    }
+
+    private static BigInteger fromBigByteArray(byte[] b) {
+        return new BigInteger(DatatypeConverter.printHexBinary(b), 16);
+    }
+
+    private static byte[] toBigByteArray(BigInteger n) {
+        byte[] b = n.toByteArray();
+        int i = 0;
+        while (b[i] == 0)
+            i++;
+        return Arrays.copyOfRange(b, i, b.length);
+    }
+
+    private static byte[] parseHexBinary(String hexString) {
+
+        if (hexString.length() % 2 != 0) {
+            hexString = '0' + hexString;
+        }
+
+        return DatatypeConverter.parseHexBinary(hexString);
+    }
+
+    private static byte[] sha1(byte[]... ba) {
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA-1");
+            for (byte[] b : ba) {
+                md.update(b);
+            }
+            return md.digest();
+        }
+        catch(NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private static byte[] pad(BigInteger n) {
+        final byte[] bn = n.toByteArray();
+        final int start = bn.length > SRP_KEY_SIZE ? bn.length - SRP_KEY_SIZE : 0;
+        return Arrays.copyOfRange(bn, start, SRP_KEY_SIZE+start);
+    }
+
+    private static BigInteger getScramble(BigInteger x, BigInteger y) {
+        return fromBigByteArray(sha1(pad(x), pad(y)));
+    }
+
+    private static BigInteger getSecret() {
+        byte[] b = new byte[SRP_KEY_SIZE/8];
+        SecureRandom random = new SecureRandom();
+        random.nextBytes(b);
+        return fromBigByteArray(b);
+    }
+
+    private static byte[] getSalt() {
+        byte[] b = new byte[SRP_SALT_SIZE];
+        SecureRandom random = new SecureRandom();
+        random.nextBytes(b);
+        return b;
+    }
+
+
+    private static BigInteger getUserHash(String user, String password, byte[] salt) {
+        final byte[] hash1 = sha1(user.getBytes(), ":".getBytes(), password.getBytes());
```

Comment:
Not entirely sure, but I think this should use UTF-8 (as in `getBytes(StandardCharsets.UTF_8)`